server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -456,22 +465,31 @@ private static void ensureToBeDecommissionedAttributeWeighedAway(ClusterState st
 
     private static void ensureEligibleRequest(
         DecommissionAttributeMetadata decommissionAttributeMetadata,
-        DecommissionAttribute requestedDecommissionAttribute
+        DecommissionRequest decommissionRequest
     ) {
-        String msg = null;
+        String msg;
+        DecommissionAttribute requestedDecommissionAttribute = decommissionRequest.getDecommissionAttribute();
         if (decommissionAttributeMetadata != null) {
             // check if the same attribute is registered and handle it accordingly
             if (decommissionAttributeMetadata.decommissionAttribute().equals(requestedDecommissionAttribute)) {
                 switch (decommissionAttributeMetadata.status()) {
-                    // for INIT and FAILED - we are good to process it again
+                    // for INIT - check if it is eligible internal retry
                     case INIT:
+                        if (decommissionRequest.id().equals(decommissionAttributeMetadata.requestID()) == false) {
+                            throw new DecommissioningFailedException(
+                                requestedDecommissionAttribute,
+                                "same request is already in status [INIT]"
+                            );
+                        }
```

Comment:
Shouldn't we check for the same request id at every state transition to ensure one and only one request can execute the request till completion?

Reply:
Added assert during INIT, will add assert once we call the drain method. Post which we don't keep the request object
```
assert decommissionAttributeMetadata.requestID().equals(decommissionRequest.id());
```